File: cpp/0053最大子数组和.cpp
#include<vector>
#include<iostream>
#include<math.h>

using namespace std;

int maxSubArray(vector<int> &nums)
{
    if(!nums.size())
        return nums[0];

    int temp = nums[0];
    int res = nums[0];

    for (int i = 1;i<nums.size();++i)
    {
        temp = max(temp + nums[i], nums[i]);
        res = max(res, temp);
    }

    return res;
}

int main()
{
    cout << max(-1, 11);
}

File: python/0053最大子数组和.py
nums=[5,4,-1,7,8]

def maxSubArrray(nums):
    if(len(nums)==1):
        return nums[0]
    
    temp=nums[0]
    res=nums[0]
    
    for i in nums[1:]:
        # 计算以nums[i]结尾的最大连续子串和
        temp=max(temp+i,i)
        
        res=max(temp,res)
    
    return res


print(maxSubArrray(nums))
    